workflow-HowTo.md

```diff
@@ -0,0 +1,229 @@
+  * The "Rebase and merge" method has no way of adding metadata to the commit (like `Co-authored-by: ...`).
+
+Therefore we **use our own local repository** to merge and push.
+
+To be able to pull branches from other repos easily, one should add them on demand, but at least `upstream`, e.g. using:
+```
+git remote add upstream git@github.com:ksh-community/ksh.git
+git remote add jghub git@github.com:*collaborator*/ksh.git
+git remote -v
+```
+Note that **upstream** is now assigned to the official ksh repository on `https://github.com/ksh-community/ksh` and **jghub** to `git@github.com:*collaborator*/ksh.git`.
+
+###	1. Sync with upstream
+Clear any `am`/`rebase` that may already be underway and sync with the upstream. This should be actually done by every ksh developer before he creates a new branch and before he rebases his branch in turn to create a PR from it:
+```
+git am --abort
+git rebase --abort
+git checkout master
+git fetch upstream
+git merge --ff-only upstream/master
+git push origin master
+```
+If the checkout above does not work because of pending changes, just stash them away using `git stash` and when done with merging, get them back using `git stash pop`.
+
+###	2. Pull the PR into a new local branch
+First, one should create a local branch, where the PR changes get merged in. Actually one may merge the changes directly into the master, however, usually one wanna **review** and **test** the changes, what sometimes takes a little bit longer or need to be postponed for this or that reason and thus would block any progress on the master or on your own branches. So with a new branch one may cherry pick, reword, edit, squash, join and drop commits, make other fine grained adjustments before the PR gets merged into the related upstream branch and if anything goes really wrong, one always has the option to simply drop the branch/changes made so far and start over again.
+
+To find out, what to merge into the new branch, one can go to the [ksh repository](https://github.com/ksh-community/ksh) page, click on the [Pull requests](https://github.com/ksh-community/ksh/pulls) tab, and select the desired PR - you will end up with an URL like `https://github.com/ksh-community/ksh/pull/123` - NOTE: the number at the end of the URL is the github assigned **ID** of the PR. On the PR page click on the [command line instructions](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) link right beside the green `Squash and merge` button. The pattern you will see is basically `git pull git://github.com/*collaborator*/ksh.git *branch*`. So for our example:
```

Comment:
This might be a bit picky, but everyone I know always says URL by reading out the letters U-R-L, which means it has to be prefixed with an 'a':
s/an URL/a URL/